### ISSUE TITLE
to_dict: decimal_type=str cause error

### ISSUE THREAD

**Issue by spumer** [brunato/xmlschema#66]
I try to convert decimal to string like in guide (Decoding to JSON section):
https://media.readthedocs.org/pdf/xmlschema/latest/xmlschema.pdf

```
self = XsdSingleFacet('minInclusive', value=Decimal('0'), fixed=False)
x = '5000'

    def min_inclusive_validator(self, x):
>       if x < self.value:
E       TypeError: '<' not supported between instances of 'str' and 'decimal.Decimal'
```

**Comment by brunato** [comment]
I think that the better solution for this is to move the optional `decimal_type` conversion at element/attribute level, avoiding modifications at schema's validators level.

So `decimal_type` will not appling anymore at type level, but It doesn't seem to be a problem for data conversion to JSON, that is usually applied at element level.

**Comment by brunato** [comment]
The fix for this is available with release 0.9.30

**Comment by spumer** [comment]
@brunato you are right about conversion to JSON at element level :)
And thanks for fix!
